examples/fox_in_a_hole/fox_in_a_hole.py

```diff
@@ -34,38 +34,90 @@
 class Game(abc.ABC):
     """Abstract ancestor class for Fox-in-a-hole game.
 
-    Parameters:
-    * hole_nr: integer - Number of holes
-    * seed:    integer - Seed for random number generated (used for testing)
+    This class does the mechanical operations of the
+    Fox-in-a-hole game:  Keeping track of the history of
+    moves, running the game by looping through guesses,
+    and printing out the history.
+
+    This class has two variants.  `ClassicalGame`
+    demonstrates how to play the classical version of
+    Fox-in-a-hole, often found in riddle and puzzle books.
+    `QuantumGame` is a quantum variant, where, instead of
+    moving from one hole to the next, the fox will move
+    to a superposition of both adjacent holes by using a
+    `Split` operator.
+
+    This class has 4 abstract methods.  These methods will
+    vary depending on whether the game is the classical or
+    quantum variant:
+
+    - initialize_state(): This function will initialize the game.
+    - state_to_string(): This returns the state of the game as
+        a string so it can be printed out.
+    - check_guess(guess):  This checks whether the user's guess
+        is correct.  In the classical version, this just checks
+        whether the fox is in the hole. In the quantum version,
+        this performs a measurement to determine if the fox is in
+        this location.
+    - take_random_move():  This function should move the fox.
+        In the classical version, the fox moves to an adjacent hole.
+        In the quantum version, the fox "splits" and moves to both
+        adjacent holes.
+
+    Args:
+        number_of_holes:  The number of holes that the fox
+            can hide in.
+        seed: Seed for random number generator. (used for testing)
     """
 
-    def __init__(self, hole_nr=5, seed=None):
+    def __init__(self, number_of_holes: int = 5, seed: int = None):
```

Comment:
`seed: Optional[int] = None` since default value is not an int

Reply:
Done.